backend/wmg/pipeline/summary_cubes/calculate_markers.py

```diff
@@ -145,12 +115,120 @@ def _query_tiledb(
     agg = query.groupby(gb_dims_es).sum()
     n_cells = cell_counts_query.groupby(gb_dims).sum()["n_total_cells"]
 
-    keep_dataset_ids = group_by_dims is None and "dataset_id" in filters.keys()
-    if genes is None:
-        genes = list(agg.index.levels[0])
+    genes = list(agg.index.levels[0])
+    n_cells_per_gene, n_cells_index = _calculate_true_n_cells(n_cells, genes, dataset_to_gene_ids, keep_dataset_ids)
+
+    # roll up across cell type descendants
+    if "cell_type_ontology_term_id" in agg.index.names:
```

Comment:
if `cell_type_ontology_term_id` is a group-by dimension, then it was present in the target population filters. In that case, we want to roll-up data across cell type descendants.